src/Makefile.am

```diff
@@ -51,3 +53,27 @@ nobase_include_HEADERS = \
 
 nobase_noinst_HEADERS = \
         ../include/opendht/request.h
+
+######################
+#  ARGON2 submodule  #
+######################
+
+noinst_LTLIBRARIES = libargon2.la
```

Comment:
I guess there may have a `src/libargon2.la` file lying in your directory. This is due to the switch between system and included argon2. To help preventing this, I'll add a `clean-local` rule.